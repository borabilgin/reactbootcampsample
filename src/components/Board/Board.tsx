import * as React from 'react';
import Square from '../Square/Square';

export interface BoardProps {
    squares: string[];
    onClick: (i: number) => any;
}

const Board: React.StatelessComponent<BoardProps> = ({squares, onClick}) => {
    // this is an example of a pure function
    const renderSquare = (i: number) => {
        return <Square value={squares[i]} onClick={() => onClick(i)} />;
    };

    return (
        <div>
            <div className="board-row">
                {/* below is a JSX expression, encapsulated in square brackets */}
                {renderSquare(0)}
                {renderSquare(1)}
                {renderSquare(2)}
            </div>
            <div className="board-row">
                {renderSquare(3)}
                {renderSquare(4)}
                {renderSquare(5)}
            </div>
            <div className="board-row">
                {renderSquare(6)}
                {renderSquare(7)}
                {renderSquare(8)}
            </div>
        </div>
    );
};

export default Board;