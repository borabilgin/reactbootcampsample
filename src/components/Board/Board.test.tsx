import * as React from 'react';
import { shallow } from 'enzyme';
import Board, { BoardProps } from './Board';
import Square from './../Square/Square';

describe('Board', () => {

    let mountedBoard: any;
    let props: BoardProps = {
        squares: [],
        onClick: () => undefined
    };

    beforeEach(() => {
        mountedBoard = shallow(<Board {...props} />);
    });

    // Not that useful of a test, but demo's how to simulate an event 
    // and use a spy callback function
    it('should call onClick prop when clicked', () => {
        const spy = jest.fn();
        props.onClick = spy;
        mountedBoard.setProps(props);
        mountedBoard.find(Square).first().simulate('click');
        expect(spy).toHaveBeenCalled();
    });

    it('should render 9 squares', () => {
        expect(mountedBoard.find(Square).length).toEqual(9);
    });

    // Snapshot test
    it('should render 9 squares', () => {
        expect(mountedBoard.find(Square).length).toMatchSnapshot();
    });

    // Snapshot of render
    it('should render 9 squares', () => {
        const wrapper = shallow(<Board {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});