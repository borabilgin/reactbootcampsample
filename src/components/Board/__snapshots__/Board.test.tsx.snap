// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render 9 squares 1`] = `9`;

exports[`Board should render 9 squares 2`] = `
<div>
  <div
    className="board-row"
  >
    <Square
      onClick={[Function]}
    />
    <Square
      onClick={[Function]}
    />
    <Square
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <Square
      onClick={[Function]}
    />
    <Square
      onClick={[Function]}
    />
    <Square
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <Square
      onClick={[Function]}
    />
    <Square
      onClick={[Function]}
    />
    <Square
      onClick={[Function]}
    />
  </div>
</div>
`;
